[inductor] Call render() once for templates

jansel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #105987

This is more code, but perhaps easier to understand?  Both @Chillee and @ipiszy expressed confusion that we rendered templates twice to reach a fixed point.  This removes that.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

jansel committed:
[inductor] Call render() once for templates

[ghstack-poisoned]

Chillee commented:
I feel like this makes more sense to me, but now that I've understood the templating system I am no longer a neutral observer :)

jansel commented:
@pytorchbot rebase

jansel committed:
Update on "[inductor] Call render() once for templates"


This is more code, but perhaps easier to understand?  Both Chillee and ipiszy expressed confusion that we rendered templates twice to reach a fixed point.  This removes that.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

jansel commented:
@pytorchbot merge
